scripts/SANS/sans/algorithm_detail/mask_workspace.py

```diff
@@ -127,10 +127,17 @@ def mask_with_mask_files(mask_info, inst_info, workspace):
         load_alg = create_unmanaged_algorithm(load_name, **load_options)
         mask_alg = create_unmanaged_algorithm("MaskDetectors")
 
-        # Masker
-        for mask_file in mask_files:
-            mask_file = find_full_file_path(mask_file)
+        file_paths = [find_full_file_path(i) for i in mask_files]
+        # Find full file path returns an empty string, so we need to remake it
+        missing_file_paths = [mask_files[i] for i, path in enumerate(file_paths) if not path]
+
+        err_str = "The following mask files are missing:"
+        err_str += "\n".join(missing_file_paths)
```

Comment:
Nitpick: is it better to create this string inside the `if` statement, as it's only in this case that it will be used?

Reply:
Good shout, originally I planned on using the err_string to determine if we should raise so that's a good shout moving it around :+1: